### ISSUE TITLE
Questions about training procedure

### ISSUE THREAD

**Issue by alderpaw** [chen700564/causalFSED#3]
Hey, thanks for your work.
How do you organize support set and query set during training? In the paper you said "We randomly sample few instances as support set and all other instances in the test set are used as queries", which means you only used the test set? I'm a little confused about this.

**Comment by chen700564** [comment]
That sentence describes the testing process. When training, we random sample support set and queries. (It can be seen in reader.py: FewEventDetection_trainbaselinereader/FewEventDetection_traincausalreader: def bert_data)

**Comment by alderpaw** [comment]
> 

Thanks for your reply.
Does that mean you organize support/query set with a positive class and N negative classes? So what's the difference with traditional N-way-K-shot setting?

**Comment by chen700564** [comment]
Yes, but there is little difference in the training process. The difference is mainly in test process (support set is from only one class and the query is the rest of sentences in the test set) and most of the previous work used different sampling strategies for training and testing (they may use 20-way 5-shot when training and 5-way 5-shot when testing)

**Comment by alderpaw** [comment]
> Yes, but there is little difference in the training process. The difference is mainly in test process (support set is from only one class and the query is the rest of sentences in the test set) and most of the previous work used different sampling strategies for training and testing (they may use 20-way 5-shot when training and 5-way 5-shot when testing)

Thanks for your patience, I understand the training and test procedure now.
And do you just concentrate on trigger classification problem rather than trigger identification? It seems that NONE event types are not considered.

**Comment by chen700564** [comment]
This is trigger identification. Tokens that are not marked as trigger words will be seen as NONE event in the sentences in  support set (so there is an addition category when training and testing, you can refer to the Appendix B in our paper.)
